Unify the version naming convention

mgol opened the pull request:
So far some browsers were just omitting a dot in the version string and some
replaced it by an underscore. This commit changes all those versions to use
the underscore translation. This will reduce confusion; with some environments
using rapidly increasing major version numbers it becomes hard to distinguish
"41" from "4.1" when both are translated to "41" in the browser name.

The only exception is Duktape which needs to have names maching keys matching
what Duktape.version reports.

Also:
### 1)
Rename Duktape shortcuts from "DUK 20" to "DUK 2.0" & "DUK 21" to "DUK 2.1"

The previous versions were incorrect. Lots of engines have high major version
numbers so version 21 sounded plausible.

### 2)
Change iOS 10.0-10.3 to 10.0-10.2

iOS 10.3 was incorrectly included in a group with 10.0-10.2.

mgol committed:
Change iOS 10.0-10.3 to 10.0-10.2

iOS 10.3 was incorrectly included in a group with 10.0-10.2.

mgol committed:
Rename Duktape shortcuts from "DUK 20" to "DUK 2.0" & "DUK 21" to "DUK 2.1"

The previous versions were incorrect. Lots of engines have high major version
numbers so version 21 sounded plausible.

mgol committed:
Unify the version naming convention

So far some browsers were just omitting a dot in the version string and some
replaced it by an underscore. This commit changes all those versions to use
the underscore translation. This will reduce confusion; with some environments
using rapidly increasing major version numbers it becomes hard to distinguish
"41" from "4.1" when both are translated to "41" in the browser name.

The only exception is Duktape which needs to have names maching keys matching
what Duktape.version reports.

ljharb reviewed (approved):
We should fix the duktape result generation as well.

mgol commented:
> We should fix the duktape result generation as well.

This means there will be no direct matching between the key we use and the one reported by `Duktape.version`. Is that OK?

ljharb commented:
`Duktape.version` reports `duktape20`? O.o

mgol commented:
Unfortunately, yes...

mgol commented:
Hmm, no, actually it seems the comment is not 100% correct. What it actually reports is:
```
duk> Duktape.version
= 20100
```
Even weirder but then if we already have to transform it we might as well add an underscore.

ljharb commented:
yeah, that's a really stupid version string; it should use semver. I see no value in trying to match that.

mgol commented:
It's not even a string, it's a number. :)

mgol commented:
@ljharb Should I change that script in this PR and unify Duktape entries as well? Or in a new one?

ljharb commented:
Let's do another commit in this pr

mgol committed:
Unify the naming convention for Duktape, fix its results script

Duktape now follows all the other engines in that version x.y is translated to
the key "duktapex_y" instead of "duktapexy".

The script comparing test results of this engine with the recorded values
has been modified to account for this change.

mgol commented:
@ljharb PR updated.

afmenez commented on environments.json:
```diff
@@ -1265,7 +1265,7 @@
       "non-standard"
     ]
   },
-  "safari51": {
```
This will conflict with my patch on #1099

ljharb replied:
Do you think it'd be easier to merge yours first, and then rebase this one? or would it be easier to merge this, and rebase yours on top?

afmenez replied:
I believe both would be about the same amount of work.  But I pushed my patch first... :)

mgol replied:
@ljharb I can wait for the @afmenez's patch to be landed & then rebase mine. I'm fine with solving those merge conflicts, I just hope I won't have to do that multiple times.

ljharb replied:
ok, sounds good

afmenez commented on environments.json:
```diff
@@ -1283,7 +1283,7 @@
     "obsolete": true
   },
-  "safari71_8": {
+  "safari7_1_8": {
```
This is not Safari 7.1.8, but Safari 7.1 & Safari 8.  Also fixed on my patch on #1099

afmenez reviewed (changes requested):
It would be really nice if you could revert the Safari changes so we can merge my more complete patch for Safari (#1099) without conflicts.

mgol commented:
@afmenez You mean Safari & iOS changes, don't you?

@ljharb Does that sound OK to you? I can revert those particular changes.

afmenez commented:
Yep, I did a larger change on Safari and iOS, adding some versions and results, and it already included all the changes you did.  If you revert just these parts, I believe both of our patches will be able to merge.  :)

mgol commented:
PR rebased.
